### PR TITLE
[Java] Concurrent Message Processing

### PR DESCRIPTION
**Pull request by MauriceVanVeen**
Resolves https://github.com/ConnectEverything/nats-by-example/issues/247

Replaces PR from fork: https://github.com/ConnectEverything/nats-by-example/pull/248

### CODE REVIEW

File: examples/messaging/concurrent/java/Main.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        String natsURL = System.getenv("NATS_URL");
+        if (natsURL == null) {
+            natsURL = "nats://127.0.0.1:4222";
+        }
+
+        // Setup options using a Dispatcher with executor.
+        // A custom executor can be specified if desired, using `.executor(..)`.
+        Options options = Options.builder()
+                .server(natsURL)
+                .useDispatcherWithExecutor()
```
Comment (scottf) [inline examples/messaging/concurrent/java/Main.java:23]: Don't use this.

**Review by scottf — CHANGES_REQUESTED** [review]
I think the intention of this was to show queue subscribing that got messages in parallel (concurrent). We might need to look at the other examples and fix them

**Comment by MauriceVanVeen** [comment]
@scottf, the description on NBE does mention queue groups, but it then goes on to say:

> However, even within a subscription, it may be desirable to handle messages concurrently. This example shows how this can be achieved in the clients that support it.

Apart from queue groups and the dispatcher with executor, not sure if the Java client supports another way for concurrent processing?

**Comment by scottf** [comment]
> Apart from queue groups and the dispatcher with executor, not sure if the Java client supports another way for concurrent processing?

I think queue group and dispatchers is the way to go for this example

**Comment by MauriceVanVeen** [comment]
Have updated it to use dispatchers+queue groups now.

**Pull request merged**